opengrok-indexer/src/test/java/org/opengrok/indexer/history/RepositoryFactoryTest.java

```diff
@@ -43,32 +53,78 @@
  * @author Vladimir Kotal
  */
 public class RepositoryFactoryTest {
-    private static TestRepository repository;
-    
+    private static RuntimeEnvironment env;
+    private static TestRepository repository = new TestRepository();
+    private static Set<String> savedDisabledRepositories;
+    private static boolean savedIsProjectsEnabled;
+
+    @Rule
+    public ConditionalRunRule rule = new ConditionalRunRule();
+
     @BeforeClass
     public static void setUpClass() throws Exception {
-        repository = new TestRepository();
-        repository.create(RepositoryFactory.class.getResourceAsStream("repositories.zip"));
+        env = RuntimeEnvironment.getInstance();
+        repository.create(RepositoryFactoryTest.class.getResourceAsStream("repositories.zip"));
+        savedDisabledRepositories = env.getDisabledRepositories();
+        savedIsProjectsEnabled = env.isProjectsEnabled();
     }
     
     @AfterClass
-    public static void tearDown() {
+    public static void tearDownClass() {
         if (repository != null) {
             repository.destroy();
             repository = null;
         }
     }
 
+    @After
+    public void tearDown() {
+        env.setRepoCmds(Collections.emptyMap());
+        env.setDisabledRepositories(savedDisabledRepositories);
+        env.setProjectsEnabled(savedIsProjectsEnabled);
+    }
+
     @ConditionalRun(RepositoryInstalled.MercurialInstalled.class)
     @Test
     public void testRepositoryMatchingSourceRoot() throws IllegalAccessException, InvocationTargetException,
             ForbiddenSymlinkException, InstantiationException, NoSuchMethodException, IOException {
 
         File root = new File(repository.getSourceRoot(), "mercurial");
-        RuntimeEnvironment env = RuntimeEnvironment.getInstance();
         env.setSourceRoot(root.getAbsolutePath());
         env.setProjectsEnabled(true);
-        assertNull(RepositoryFactory.getRepository(root));
+        Repository repo = RepositoryFactory.getRepository(root);
+        assertNull("should not get repo for root if projects enabled", repo);
+    }
+
+    @ConditionalRun(RepositoryInstalled.MercurialInstalled.class)
+    @Test
+    public void testNormallyEnabledMercurialRepository() throws IllegalAccessException,
+            InvocationTargetException, ForbiddenSymlinkException, InstantiationException,
+            NoSuchMethodException, IOException {
+
+        File root = new File(repository.getSourceRoot(), "mercurial");
+        env.setSourceRoot(root.getAbsolutePath());
+        assertNotNull("should get MercurialRepository",
```

Comment:
nit: the message describes different assertion than the one made here

Reply:
Isn't think OK since the message is an expectation description — i.e. "what _should_ have happened?" — displayed on failure?

Reply:
As I said this is probably nit picking, however the test verifies the non-null property rather than checking that the returned object is actually an instance of `MercurialRepository`.